hack/netpol/pkg/utils/networkpolicyspecbuilder.go

```diff
@@ -0,0 +1,135 @@
+	Namespace string
+}
+
+func (n *NetworkPolicySpecBuilder) Get() *networkingv1.NetworkPolicy {
+	return &networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      n.Name,
+			Namespace: n.Namespace,
+		},
+		Spec: n.Spec,
+	}
+}
+
+func (n *NetworkPolicySpecBuilder) SetPodSelector(labels map[string]string) *NetworkPolicySpecBuilder {
+	ps := metav1.LabelSelector{
+		MatchLabels: labels,
+	}
+	n.Spec.PodSelector = ps
+	return n
+}
+
+func (n *NetworkPolicySpecBuilder) SetName(namespace string, name string) *NetworkPolicySpecBuilder {
+	n.Namespace = namespace
+	n.Name = name
+	return n
+}
+
+func (n *NetworkPolicySpecBuilder) AddIngress(protoc *v1.Protocol, port *int, portName *string, cidr *string, podSelector map[string]string, nsSelector map[string]string, podSelectorMatchExp *[]metav1.LabelSelectorRequirement, nsSelectorMatchExp *[]metav1.LabelSelectorRequirement) *NetworkPolicySpecBuilder {
+
+	var ps *metav1.LabelSelector
+	var ns *metav1.LabelSelector
+	if n.Spec.Ingress == nil {
+		n.Spec.Ingress = []networkingv1.NetworkPolicyIngressRule{}
+	}
+
+	if podSelector != nil {
+		ps = &metav1.LabelSelector{
+			MatchLabels: podSelector,
+		}
+	}
+	if nsSelector != nil {
+		ns = &metav1.LabelSelector{
+			MatchLabels: nsSelector,
+		}
+	}
+
+	var policyPeer []networkingv1.NetworkPolicyPeer
+	if ps != nil || ns != nil {
+		policyPeer = []networkingv1.NetworkPolicyPeer{{
+			PodSelector:       ps,
+			NamespaceSelector: ns,
+			IPBlock:           nil, // todo add in IPBlock support
+		}}
+	}
+
+	var ports []networkingv1.NetworkPolicyPort
+	if port != nil && portName != nil {
+		panic("specify portname or port, not both")
+	}
+	if port != nil {
+		ports = []networkingv1.NetworkPolicyPort{
+			{
+				Port: &intstr.IntOrString{IntVal: int32(*port)},
+			},
+		}
+	}
+	if portName != nil {
+		ports = []networkingv1.NetworkPolicyPort{
+			{
+				Port: &intstr.IntOrString{Type: intstr.String, StrVal: *portName},
+			},
+		}
+	}
+	newRule := networkingv1.NetworkPolicyIngressRule{
+		From:  policyPeer,
+		Ports: ports,
+	}
+	n.Spec.Ingress = append(n.Spec.Ingress, newRule)
+	return n
+}
+
+// AddEgressDNS mutates the nth policy rule to allow DNS, convenience method
+func (n *NetworkPolicySpecBuilder) WithEgressDNS() *NetworkPolicySpecBuilder {
+	protocolUDP := v1.ProtocolUDP
+	route53 := networkingv1.NetworkPolicyPort{
+		Protocol: &protocolUDP,
+		Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: 53},
+	}
+
+	for _, e := range n.Spec.Egress {
+		e.Ports = append(e.Ports, route53)
+	}
+	return n
+}
+
+func (n *NetworkPolicySpecBuilder) AddEgress(protoc *v1.Protocol, port *int, portName *string, cidr *string, podSelector map[string]string, nsSelector map[string]string, podSelectorMatchExp *[]metav1.LabelSelectorRequirement, nsSelectorMatchExp *[]metav1.LabelSelectorRequirement) *NetworkPolicySpecBuilder {
+	// For simplicity, we just reuse the Ingress code here.  The underlying data model for ingress/egress is identical
```

Comment:
i guess AddIngress and AddEgress can share some common code via a internal func

Reply:
yup, didnt bother abstracting that away but they could if we want to keep this.  Im assuming we may decided a Struct is better for all of these functions later on, hence i didnt refactor it yet.